mmv1/third_party/tgc_next/test/assert_test_files.go

```diff
@@ -0,0 +1,281 @@
+	if err != nil {
+		return fmt.Errorf("error when writing the file %s", roundtripTfFilePath)
+	}
+	defer os.Remove(roundtripTfFilePath)
+
+	if diff := cmp.Diff(string(roundtripConfigData), string(exportConfigData)); diff != "" {
+		log.Printf("Roundtrip config is different from the export config.\nroundtrip config:\n%s\nexport config:\n%s", string(roundtripConfigData), string(exportConfigData))
+		return fmt.Errorf("Test %s got diff (-want +got): %s", testName, diff)
+	}
+
+	return nil
+}
+
+// Gets the ancestry cache for tfplan2cai conversion
+func getAncestryCache(assets []caiasset.Asset) map[string]string {
+	ancestryCache := make(map[string]string, 0)
+
+	for _, asset := range assets {
+		ancestors := asset.Ancestors
+		if len(ancestors) != 0 {
+			var path string
+			for i := len(ancestors) - 1; i >= 0; i-- {
+				curr := ancestors[i]
+				if path == "" {
+					path = curr
+				} else {
+					path = fmt.Sprintf("%s/%s", path, curr)
+				}
+			}
+
+			if _, ok := ancestryCache[ancestors[0]]; !ok {
+				ancestryCache[ancestors[0]] = path
+			}
+
+			project := utils.ParseFieldValue(asset.Name, "projects")
+			projectKey := fmt.Sprintf("projects/%s", project)
+			if strings.HasPrefix(ancestors[0], "projects") && ancestors[0] != projectKey {
+				if _, ok := ancestryCache[projectKey]; !ok {
+					ancestryCache[projectKey] = path
+				}
+			}
+		}
+	}
+	return ancestryCache
+}
+
+// Compares HCL and finds all of the keys in map1 are in map2
+func compareHCLFields(map1, map2 map[string]interface{}, path string, ignoredFields map[string]bool) []string {
+	var missingKeys []string
+	for key, value1 := range map1 {
+		if value1 == nil {
+			continue
+		}
+
+		currentPath := path + "." + key
+		if path == "" {
+			currentPath = key
+		}
+
+		if ignoredFields[currentPath] {
+			continue
+		}
+
+		value2, ok := map2[key]
+		if !ok || value2 == nil {
+			missingKeys = append(missingKeys, currentPath)
+			continue
+		}
+
+		switch v1 := value1.(type) {
+		case map[string]interface{}:
+			v2, _ := value2.(map[string]interface{})
+			missingKeys = append(missingKeys, compareHCLFields(v1, v2, currentPath, ignoredFields)...)
+		case []interface{}:
+			v2, _ := value2.([]interface{})
+
+			for i := 0; i < len(v1); i++ {
+				nestedMap1, ok1 := v1[i].(map[string]interface{})
+				nestedMap2, ok2 := v2[i].(map[string]interface{})
+				if ok1 && ok2 {
+					keys := compareHCLFields(nestedMap1, nestedMap2, fmt.Sprintf("%s[%d]", currentPath, i), ignoredFields)
+					missingKeys = append(missingKeys, keys...)
+				}
+			}
+		default:
+		}
+	}
+
+	return missingKeys
+}
+
+// Converts a tfplan to CAI asset, and then converts the CAI asset into HCL
+func getRoundtripConfig(t *testing.T, testName string, tfDir string, ancestryCache map[string]string, logger *zap.Logger) ([]byte, error) {
+	fileName := fmt.Sprintf("%s_export", testName)
+
+	// Run terraform init and terraform apply to generate tfplan.json files
+	terraformWorkflow(t, tfDir, fileName)
```

Comment:
I am trying to remove `t *testing.T` from the signature of the functions and return error. 

The function `terraformWorkflow` needs the same modification. I will do that in a separate PR later and then remove  `t *testing.T` from the signature of the functions.